st2actions/st2actions/notifier/notifier.py

```diff
@@ -167,7 +167,15 @@ def _post_notify_subsection_triggers(self, liveaction=None, execution=None,
             payload['execution_id'] = execution_id
             payload['status'] = liveaction.status
             payload['start_timestamp'] = isotime.format(liveaction.start_timestamp)
-            payload['end_timestamp'] = isotime.format(liveaction.end_timestamp)
+
+            try:
+                payload['end_timestamp'] = isotime.format(liveaction.end_timestamp)
+            except AttributeError:
+                # This can be raised if liveaction.end_timestamp is None, which is caused
+                # when policy cancels a request due to concurrency
+                # In this case, use the start_timestamp instead
+                payload['end_timestamp'] = isotime.format(liveaction.start_timestamp)
```

Comment:
I would personally just set it to the time when the execution was canceled / this method is called (aka now). I know this is not the most accurate, but yeah, other alternative is leaving it empty which is probably the most correct since execution never started.